src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -58,20 +58,28 @@ public class DataBlockService : IDataBlockService
 
                     await _context.SaveChangesAsync();
 
-                    return await GetAsync(added.Id);
+                    return await Get(added.Id);
                 });
         }
 
-        public async Task<Either<ActionResult, Unit>> DeleteAsync(Guid releaseId, Guid id)
+        public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId, Guid id)
         {
             return await _persistenceHelper
-                .CheckEntityExists<DataBlock>(id)
-                .OnSuccess(CheckCanUpdateReleaseForDataBlock)
-                .OnSuccess(block => GetDeleteDataBlockPlan(releaseId, id)
-                .OnSuccessVoid(async deletePlan =>
-                {
-                    await DeleteDataBlocks(deletePlan);
-                }));
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(
+                    async release => await _persistenceHelper
+                        .CheckEntityExists<DataBlock>(id)
+                        .OnSuccess(CheckCanUpdateReleaseForDataBlock)
+                        .OnSuccess(
+                            block => GetDeletePlan(release.Id, id)
+                                .OnSuccessVoid(
```

Comment:
Can this be written `OnSuccessVoid(DeleteDataBlocks)` or `OnSuccessDo(DeleteDataBlocks)`?  Do they have variants that successfully take a Task?

Reply:
Yep it can, I've changed it to `OnSuccessVoid(DeleteDataBlocks)` :+1: